controllers/flinkcluster_converter.go

```diff
@@ -605,18 +586,37 @@ func getJobManagerIngressHost(ingressHostFormat string, clusterName string) stri
 	return jobManagerIngressHostRegex.ReplaceAllString(ingressHostFormat, clusterName)
 }
 
-func isStopDelayExpired(
-	clusterStatus v1alpha1.FlinkClusterStatus,
-	delayMinutes int32,
-	now time.Time) bool {
-	if clusterStatus.State != v1alpha1.ClusterState.Stopping ||
-		len(clusterStatus.LastUpdateTime) == 0 {
+// Checks whether the component should be deleted after job finishes, returns
+// false if the cluster is a session cluster or the job hasn't finished yet.
+func shouldDeleteAfterJobFinishes(
+	cluster *v1alpha1.FlinkCluster, component string) bool {
+	var jobStatus = cluster.Status.Components.Job
+
+	// Session cluster.
+	if jobStatus == nil {
 		return false
 	}
-	var tc = &TimeConverter{}
-	var lastUpdateTime = tc.FromString(clusterStatus.LastUpdateTime)
-	return now.After(
-		lastUpdateTime.Add(time.Duration(delayMinutes) * time.Minute))
+
+	// Job hasn't finished yet.
+	if jobStatus.State != v1alpha1.JobState.Succeeded &&
```

Comment:
This is confused with the method name ...AfterJobFinishes?

Reply:
Updated.